packages/database/src/modelClasses/DashboardRelation.js

```diff
@@ -28,6 +28,25 @@ export class DashboardRelationType extends DatabaseType {
   ];
 }
 
+const assertRelationAttributesMatchEntity = (entityAttribute, attributeFilter) => {
+  if (!attributeFilter) return true;
+  const comparisonValue = (
+    Array.isArray(attributeFilter) ? attributeFilter : [attributeFilter]
+  ).map(v => String(v).toLowerCase());
+
+  if (comparisonValue.includes('no')) {
+    return entityAttribute === 'no' || !entityAttribute;
+  }
+  return comparisonValue.includes(entityAttribute);
+};
+
+const filterByAttributes = (entity, relation) => {
+  if (!relation.attributes_filter || !Object.keys(relation.attributes_filter).length) return true;
+  return Object.entries(relation.attributes_filter).every(([key, value]) => {
+    return assertRelationAttributesMatchEntity(entity.attributes?.[key], value);
+  });
+};
```

Comment:
I think we could rely on the database level filtering to determine if the entity matches the attributes filter. I'd lean towards that as it's always nice to not have to implement another custom filter.

Perhaps a function like this that sits on the `DashboardRelationType` itself, that looks something like:
```
  async matchesAttributeFilter(entity) {
    const { attributes_filter: attributesFilter } = this;

    const attributesDbFilter = Object.fromEntries(
      Object.entries(attributesFilter).map(([key, value]) => [`attributes->>${key}`, value]),
    );

    const filteredEntity = await this.otherModels.entity.findOne({
      id: entity.id,
      ...attributesDbFilter,
    });

    return isNotNullish(filteredEntity);
  }
```